Firestore/Source/Local/FSTLevelDB.mm

```diff
@@ -290,6 +298,18 @@ - (instancetype)initWithDirectory:(Path)directory serializer:(FSTLocalSerializer
   return self;
 }
 
+- (size_t)byteSize {
+  size_t count = 0;
+  auto iter = util::DirectoryIterator::Create(_directory);
+  for (; iter->Valid(); iter->Next()) {
+    off_t fileSize = util::FileSize(iter->file()).ValueOrDie();
+    count += fileSize;
```

Comment:
Strictly speaking this isn't safe: `size_t` is a type that measures the sizes of objects in memory, so on 32-bit systems it will be 32 bits. `off_t` is the size of files on disk, and can be defined to be 64 bits even on 32 bit systems (e.g. by passing `-D_FILE_OFFSET_BITS=64` to the build).

However, I'm not sure I care about supporting local caches larger than 4GB on 32 bit systems and I can't see us exposing the end user control of the cache size in terms of `off_t` so it seems silly to internally represent the cache size that way only to knee cap it through the control knob.

Maybe just assert we don't overflow here?

Reply:
I switched the accumulator to `off_t` and assert that it is less than `SIZE_MAX` before returning.